tests/test_utils.py

```diff
@@ -306,3 +313,513 @@ def check_multistream_classic(fnames):
+                                    color=compare_data_dict[key]['color'],
+                                    linestyle=compare_data_dict[key]
+                                    ['linestyle'],
+                                    zorder=compare_data_dict[key]['zorder'])
+
+                    axarr[i].legend(loc='upper left')
+                    axarr[i].set_ylabel('%s ($W/{m^2}$)' %variable_names[i],
+                                        size=fs)
+                    axarr[i].set_xlabel('Time of Day (Hour)', size=fs)
+                    axarr[i].set_xlim([0,24])
+                    axarr[i].xaxis.set_ticks(np.arange(0, 24, 3))
+
+                # save plot
+                plotname = '%s_%s.png' % (lat, lng)
+                os.makedirs(os.path.join(plot_dir, 'annual_mean'),
+                                        exist_ok=True)
+                savepath = os.path.join(plot_dir, 'annual_mean', plotname)
+                plt.savefig(savepath, bbox_inches='tight', dpi=dpi)
+                print("saved figure to %s" %savepath)
+
+                plt.clf()
+                plt.close()
+
+            if 'monthly_mean_diurnal_cycle' in plots_to_make:
+
+                # make monthly mean diurnal cycle plots
+                f, axarr = plt.subplots(4, 12, figsize=(35,7), sharex=True,
+                                        sharey=True)
+
+                months = ['January', 'February', 'March', 'April', 'May',
+                            'June', 'July', 'August', 'September',
+                            'October', 'November', 'December']
+
+                for i, vic_var in enumerate(vic_vars):
+                    for j in range(12):
+
+                        # calculate monthly mean diurnal cycle
+                        monthly_mean = {}
+                        for (key, df) in data.items():
+                            monthly_mean[key] = pd.DataFrame(df[vic_var].
+                                                    groupby([df['MONTH'],
+                                                    df['HOUR']]).mean())
+
+                        df = pd.DataFrame({key: d[vic_var] for key, d in
+                                        monthly_mean.items() if vic_var in d})
+
+                        for key, series in df.iteritems():
+                            series[j+1].plot(linewidth=compare_data_dict[key]
+                                        ['linewidth'],
+                                        ax=axarr[i,j],
+                                        color=compare_data_dict[key]
+                                        ['color'],
+                                        linestyle=compare_data_dict[key]
+                                        ['linestyle'],
+                                        zorder=compare_data_dict[key]['zorder'])
+
+                        if i == 0 and j == 11:
+                            axarr[i,j].legend(loc='center left',
+                                                bbox_to_anchor=(1, 0.5))
```

Comment:
I think it would clearer if you just wrote (outside of the for loops you are in:
```axarr[0, -1].legend(loc='center left', bbox_to_anchor=(1, 0.5))```